### PR TITLE
meta: ensure Application passthrough is scrubbed for snap.yaml

### PR DESCRIPTION
**Pull request by cjp256**
The original `passthrough` was leaking through into the snap.yaml
from the app_dict.  Ensure "passthrough" is popped from the dict
used to create the snap.yaml.

Add unit tests for passthrough coverage of Application.

LP #1862197

Signed-off-by: Chris Patterson <chris.patterson@canonical.com>

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

**Comment by cjp256** [comment]
Added spread test coverage for passthrough on apps, hooks, and snaps.

File: tests/spread/general/passthrough/expected_snap.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: passthrough-tests
+version: '1.0'
+summary: passthrough-tests
+description: passthrough-tests
+apps:
+  passthrough-none:
+    command: test-cmd
+    command-chain:
+    - snap/command-chain/snapcraft-runner
+    passthrough:
```
Comment (sergiusens) [inline tests/spread/general/passthrough/expected_snap.yaml]: did you miss this one?

Comment (cjp256) [reply tests/spread/general/passthrough/expected_snap.yaml]: fail  :man_facepalming:

Comment (cjp256) [reply tests/spread/general/passthrough/expected_snap.yaml]: OK well I don't feel too bad, I just copied the wrong thing at the last minute without looking too hard... whoops!

Cleaned up the test a bit too while I'm at it.

**Pull request merged**